pkg/operator/operator.go

```diff
@@ -472,9 +472,24 @@ func (o *Operator) sync(key string) error {
 		klog.Errorf("error occurred while setting status to in progress: %v", err)
 	}
 
-	taskName, err := tl.RunAll()
-	if err != nil {
-		o.reportError(err, taskName)
+	taskErrors := tl.RunAll()
+	if len(taskErrors) > 0 {
+		err := errors.New("Cluster monitoring update failed.")
+		var failedTask string
+		var tErrors []error
+		if len(taskErrors) == 1 {
+			failedTask = strings.Join(strings.Fields(taskErrors[0].Name+"Failed"), "")
+		} else {
+			failedTask = "MultipleTasksFailed"
+		}
+
+		for _, val := range taskErrors {
+			klog.Infof("Updating ClusterOperator status to failed. Err: %v", val.Err)
+			tErrors = append(tErrors, val.Err)
+		}
```

Comment:
The errors are already logged in `reportError()` so this isn't needed anymore.

Reply:
I have not understood the comment, this is adding all the errors to tErrors which is then sent to reportError(), If this is not done there will nothing to report.